docs/create-a-new-angular-component.md

```diff
@@ -0,0 +1,508 @@
+
+## 1. Create a new Angular library
+
+To create a UI library within `challenge-registry`, run:
+
+```console
+nx g @nrwl/angular:lib <new library name> --directory challenge-registry
+```
+ 
+(Optional but recommended) Use `--dryrun` to first see what and where the entities will be created;
+this will help visualize and validate the intended directory structure, e.g.
+
+```console
+$ nx g @nrwl/angular:lib awesome-lib --directory challenge-registry --dry-run
+
+>  NX  Generating @nrwl/angular:library
+
+...
+UPDATE workspace.json
+CREATE libs/challenge-registry/awesome-lib/README.md
+CREATE libs/challenge-registry/awesome-lib/tsconfig.lib.json
+CREATE libs/challenge-registry/awesome-lib/tsconfig.spec.json
+CREATE libs/challenge-registry/awesome-lib/src/index.ts
+CREATE libs/challenge-registry/awesome-lib/src/lib/challenge-registry-awesome-lib.module.ts
+CREATE libs/challenge-registry/awesome-lib/tsconfig.json
+CREATE libs/challenge-registry/awesome-lib/project.json
+UPDATE tsconfig.base.json
+CREATE libs/challenge-registry/awesome-lib/jest.config.ts
+CREATE libs/challenge-registry/awesome-lib/src/test-setup.ts
+CREATE libs/challenge-registry/awesome-lib/.eslintrc.json
+
+NOTE: The "dryRun" flag means no changes were made.
+```
+
+Due to how the Challenge Registry app is currently structured, some additional
+steps are required:
+
+1. Discard/undo changes made to `project.json` in other existing folders outside of the new library
+folder, e.g. `apps/challenge-api-gateway/project.json`. The only files you need staged are:
+
+    * files created for the new library, e.g. `libs/challenge-registry/<new library name/*`
+    * `tsconfig.base.json`
+    * `workspace.json`
+
+    You could do something like this:
+    
+    ```
+    git add libs/challenge-registry/<new library name>/ tsconfig.base.json workspace.json
+    git checkout -- .
+    ```
+
+
+2. Remove `challenge-registry-` from the filename of the module TypeScript in `src/lib/`, e.g.
+    
+      `challenge-registry-awesome-lib.module.ts` → `awesome-lib.module.ts`
+
+3. Simiarly, in `src/index.ts`, remove `challenge-registry-` from the import filepath.
+4. In the library module (`<new library name>.module.ts`), remove `ChallengeRegistry` from 
+the class name, e.g.
+    
+      `export class ChallengeRegistryAwesomeLibModule {}` → `export class AwesomeLibModule {}`
+    
+5. While in the library module, import the UI and routing modules:
+
+    ```ts
+    ...
+    import { UiModule } from '@sagebionetworks/challenge-registry/ui';
+    import { RouterModule, Routes } from '@angular/router';
+    
+    const routes: Routes = [{ path: '', component: <component> }];
+    
+    @NgModule({
+      imports: [CommonModule, RouterModule.forChild(routes), UiModule],
+      ...
+    ```
+    
+    where `<component>` will be created in the next step.
+
+> **Note**: still have questions about libraries?  See [Libraries] for more details.
```

Comment:
When I went through the steps, here's the only part confused me a bit since it requires the component to be generated first. I would consider to remove it or move it to latter steps after the component has been created - but not a big deal.

Reply:
That makes sense to me!  Moving this change to after the component is created is way more straightforward.